src/Sys/esp32Tools.cpp

```diff
@@ -0,0 +1,250 @@
+  char * loop_taskname = pcTaskGetTaskName(loop_taskHandle);    // ask for name of the known task (to make sure we are still looking at the right one)
+
+  if ((loop_taskHandle == NULL) || (loop_taskname == NULL) || (strncmp(loop_taskname, "loopTask", 8) != 0)) {
+    loop_taskHandle = xTaskGetHandle("loopTask");              // need to look for the task by name. FreeRTOS docs say this is very slow, so we store the result for next time
+  }
+
+  if (loop_taskHandle != NULL) return uxTaskGetStackHighWaterMark(loop_taskHandle); // got it !!
+  else return -1;
+}
+
+int sysTools_get_webserver_maxStackUsage(void) {
+  static TaskHandle_t tcp_taskHandle = NULL;                   // to store the task handle for later calls
+  char * tcp_taskname = pcTaskGetTaskName(tcp_taskHandle);     // ask for name of the known task (to make sure we are still looking at the right one)
+
+  if ((tcp_taskHandle == NULL) || (tcp_taskname == NULL) || (strncmp(tcp_taskname, "async_tcp", 9) != 0)) {
+    tcp_taskHandle = xTaskGetHandle("async_tcp");              // need to look for the task by name. FreeRTOS docs say this is very slow, so we store the result for next time
+  }
+
+  if (tcp_taskHandle != NULL) return uxTaskGetStackHighWaterMark(tcp_taskHandle); // got it !!
+  else return -1;
+}
+#else
+#warning cannot query task stacksize on your system
+int sysTools_get_arduino_maxStackUsage(void) { return -1;}
+int sysTools_get_webserver_maxStackUsage(void)  { return -1;}
+#endif
+
+
+// helper fuctions
+static int getCoreResetReason(int core) {
+  if (core >= ESP.getChipCores()) return 0;
+  return((int)rtc_get_reset_reason(core));
+}
+
+static String resetCode2Info(int reason) {
+  switch(reason) {
+
+    case 1 : //  1 =  Vbat power on reset
+      return F("power-on"); break;
+    case 2 : // 2 = this code is not defined on ESP32
+      return F("exception"); break;
+    case 3 : // 3 = Software reset digital core
+       return F("SW reset"); break;
+    case 12: //12 = Software reset CPU
+       return F("SW restart"); break;
+    case 5 : // 5 = Deep Sleep wakeup reset digital core
+       return F("wakeup"); break;
+    case 14:  //14 = for APP CPU, reset by PRO CPU
+      return F("restart"); break;
+    case 15: //15 = Reset when the vdd voltage is not stable (brownout)
+      return F("brown-out"); break;
+
+    // watchdog resets
+    case 4 : // 4 = Legacy watch dog reset digital core
+    case 6 : // 6 = Reset by SLC module, reset digital core
+    case 7 : // 7 = Timer Group0 Watch dog reset digital core
+    case 8 : // 8 = Timer Group1 Watch dog reset digital core
+    case 9 : // 9 = RTC Watch dog Reset digital core
+    case 11: //11 = Time Group watchdog reset CPU
+    case 13: //13 = RTC Watch dog Reset CPU
+    case 16: //16 = RTC Watch dog reset digital core and rtc module
+    case 17: //17 = Time Group1 reset CPU
+      return F("watchdog"); break;
```

Comment:
Is F() still needed? 

As:
"const" and literal strings will normally be put in flash memory by default.

?

Reply:
Your right, will change 👍